api/src/main/java/com/datastrato/gravitino/rel/Table.java

```diff
@@ -58,4 +58,15 @@ default String comment() {
   default Map<String, String> properties() {
     return Collections.emptyMap();
   }
+
+  /**
+   * Table method for working with partitions. If the table does not support partitions, an {@link
+   * UnsupportedOperationException} is thrown.
+   *
+   * @return The partition support table.
+   * @throws UnsupportedOperationException If the table does not support partition operations.
+   */
+  default SupportsPartitions supportPartitions() throws UnsupportedOperationException {
```

Comment:
How should developers use SupportsPartition if load a table? could you give a example?

Reply:
they must load a table firstly:
```
Table table = catalog.asTableCatalog().loadTable(tableIdent)
```
then, if they want to list all partition names of the table:
```
table.supportPartitions().listPartitionNames()
```